pulser-core/pulser/sampler/samples.py

```diff
@@ -167,26 +166,26 @@ def extend_duration(self, new_duration: int) -> ChannelSamples:
         if extension < 0:
             raise ValueError("Can't extend samples to a lower duration.")
 
-        new_amp = np.pad(self.amp, (0, extension))
+        new_amp = pm.pad(self.amp, (0, extension))
         # When in EOM mode, we need to keep the detuning at detuning_off
         if self.eom_blocks and self.eom_blocks[-1].tf is None:
-            final_detuning = self.eom_blocks[-1].detuning_off
+            final_detuning = float(self.eom_blocks[-1].detuning_off)
```

Comment:
I see that you regularly do `float(AbstractArray)`. From what I get, it calls `AbstractArray.astype` with `dtype=float`. In the case where a torch tensor is in `AbstractArray`, it converts it using `Array.to(dtype=float)`, but I am wondering if that effectively works, since I think it should be  `Array.to(dtype=torch.float)` (see [here](https://pytorch.org/docs/stable/generated/torch.Tensor.to.html#torch.Tensor.to))

[Edit] That's silly, I just got that it calls AbstractArray.__float__ that only handles AbstractArray._array. So it's fine by me.